cmd/waku/server/rest/lightpush_rest.go

```diff
@@ -0,0 +1,68 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/waku-org/go-waku/waku/v2/node"
+	"github.com/waku-org/go-waku/waku/v2/protocol/lightpush"
+	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"go.uber.org/zap"
+)
+
+const routeLightPushV1Messages = "/lightpush/v1/message"
+
+type LightpushService struct {
+	node *node.WakuNode
+	log  *zap.Logger
+}
+
+func NewLightpushService(node *node.WakuNode, m *chi.Mux, log *zap.Logger) *LightpushService {
+	serv := &LightpushService{
+		node: node,
+		log:  log.Named("lightpush"),
+	}
+
+	m.Post(routeLightPushV1Messages, serv.postMessagev1)
+
+	return serv
+}
+
+func (msg lightpushRequest) Check() error {
+	if msg.Message == nil {
+		return errors.New("waku message is required")
+	}
+	return nil
+}
+
+type lightpushRequest struct {
+	PubSubTopic string          `json:"pubsubTopic"`
+	Message     *pb.WakuMessage `json:"message"`
+}
+
+// handled error codes are 200, 400, 500, 503
+func (serv *LightpushService) postMessagev1(w http.ResponseWriter, req *http.Request) {
+	msg := &lightpushRequest{}
+	decoder := json.NewDecoder(req.Body)
+	if err := decoder.Decode(msg); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	defer req.Body.Close()
+
+	if err := msg.Check(); err != nil {
+		writeErrOrResponse(w, err, http.StatusBadRequest)
```

Comment:
Do note that `writeErrOrResponse` adds a `"Content-Type", "application/json; charset=UTF-8"` , while the yaml specifies `text/plain` for the responses. Might be a good moment to refactor the `writeErrOrResponse` functions to accept different headers/statuscodes (maybe via options pattern?)